Add bulk job management actions back to Retries, Dead, and Scheduled tabs

deve1212 opened the pull request:
<img width="2632" height="1562" alt="image" src="https://github.com/user-attachments/assets/97488b47-1980-4a22-9dc8-1c5cc6c4f6c3" />



### Summary
This PR adds three additional bulk action buttons — **Retry All**, **Delete All**, and **Kill All** — to the Retries page in the Sidekiq Web UI.  

These buttons provide a more efficient way to manage large retry queues without manually selecting all jobs or using the console.

### Details
- Added three buttons in the existing form alongside "Retry Now", "Delete", and "Kill".
- Each button uses `formaction` to target the existing endpoints:
  - `/retries/all/retry`
  - `/retries/all/delete`
  - `/retries/all/kill`
- Maintains existing CSRF protection and styling (`btn btn-primary` / `btn btn-danger`).
- No JavaScript or dynamic “Selected vs All” behavior added — this version focuses on restoring the previously requested bulk action buttons.

### Motivation
Restores functionality that multiple users have requested in issue discussions (e.g. [#6705](https://github.com/mperham/sidekiq/issues/6733)) and recent GitHub comments where the **Retry All** and **Delete All** buttons were found useful, especially during large retry queue operations or platform outages (like Heroku downtime).

### Testing
- Verified buttons correctly trigger existing bulk endpoints.
- Confirmed CSRF token and form submission are preserved.
- Checked visual layout — all six buttons display on one row with minimal CSS changes.

### Future improvements
- Could later add dynamic labels (“Retry Selected” vs “Retry All”) when filters or checkboxes are active.
- Potential refactor to unify form actions or enhance accessibility.

mperham commented on web/assets/stylesheets/style.css:
```diff
@@ -454,6 +454,10 @@ canvas + .table_container {
 }
 
+.bulk-retry-buttons.retry-all-button {
+  margin-left: auto;
```
"left" and "right" does not work since the Sidekiq UI supports BiDi. You need to use start and end.

https://www.tutorialpedia.org/blog/bidtext-css/#4-best-practices

deve1212 replied:
updated the css with margin-inline-start: auto;

deve1212 commented:
> Do we want similar functionality on the Dead or Scheduled tabs?

@mperham thanks for the input have added the following changes for this , please review it once and let me know-
- Added functionality for “Retry All” and “Delete All” actions in the Dead Jobs tab. 
- Added functionality for “Add All to Queue” and “Delete All” actions in the Scheduled Jobs tab.
- Introduced new APIs for bulk operations in scheduled jobs:
  POST /scheduled/all/add_to_queue
  POST /scheduled/all/delete
Both the Scheduled Jobs and Dead Jobs updates use the same JS function to toggle button visibility (similar to the previously added bulk retry functionality)

attaching images for better reference----
<img width="1511" height="761" alt="Screenshot 2025-10-20 at 1 11 50 PM" src="https://github.com/user-attachments/assets/ef1d5621-0d21-4ef0-98de-5eccd2aad1ce" />
<img width="1511" height="761" alt="Screenshot 2025-10-20 at 1 11 45 PM" src="https://github.com/user-attachments/assets/902a7262-8060-4ea6-b977-6865c5590b63" />

deve1212 commented on lib/sidekiq/web/application.rb:
```diff
@@ -318,6 +318,16 @@ def initialize(inst)
+      end
+
+      post '/scheduled/all/add_to_queue' do
+        Sidekiq::ScheduledSet.new.retry_all
```
fixed , introduced new method add_all_to_queue for scheduled jobs

deve1212 commented:
> Please add tests and the CSS needs to be refactored to remove redundancy.

@mperham As suggested, I’ve removed redundant CSS by introducing common classes for all tabs. Also added RSpec tests for the scheduled#add_all_to_queue and scheduled#delete_all APIs. The changes have been tested successfully, and the full test suite passes.

chaadow commented:
@mperham @deve1212 Hi! thanks for restoring those. However is it possible to add a confirm dialog when pressing `delete all`

I've just, intentionnally, pushed the button and it deleted everything instantly. However people sometimes can make mistakes, and the action is quite dangerous

Thanks in advance
